int32 id
float32 x_pos
float32 y_pos
float32 vel
float32 heading
